.github/workflows/ios.yml

```diff
@@ -5,16 +5,62 @@ on:
     branches: [ master ]
   pull_request:
     branches: [ master ]
-
+  
 jobs:
+
+  lint:
+    name: Lint with SwiftLint
+    runs-on: macos-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: SwiftLint report
+        run: swiftlint lint --reporter github-actions-logging --config "FlipperZero/.swiftlint.yml"
+  
   build:
     name: Build and Test FlipperZero(ios) scheme using iPhone 11 simulator
     runs-on: macos-latest
-
     steps:
       - name: Checkout
         uses: actions/checkout@v2
       - name: Xcode version selection
         run: sudo xcode-select --switch /Applications/Xcode_12.app
       - name: Xcode test
-        run: xcodebuild -sdk iphonesimulator -workspace ./FlipperZero/FlipperZero.xcodeproj/project.xcworkspace -scheme 'FlipperZero(iOS)' test DSTROOT=./output/test/build.dst OBJROOT=./output/test/build.obj SYMROOT=./output/test/build.sym SHARED_PRECOMPS_DIR=./output/test/build.pch -destination platform='iOS Simulator,name=iPhone 11' | xcpretty
+        run: xcodebuild -sdk iphonesimulator -workspace ./FlipperZero/FlipperZero.xcodeproj/project.xcworkspace -scheme 'FlipperZero(iOS)' -derivedDataPath output/DerivedData -enableCodeCoverage YES test DSTROOT=output/test/build.dst OBJROOT=output/test/build.obj SYMROOT=output/test/build.sym SHARED_PRECOMPS_DIR=./output/test/build.pch -destination platform='iOS Simulator,name=iPhone 11' | xcpretty
+      - name: Export test coverege
+        run: |
+          mkdir output/coverage
+          if [ ${GITHUB_REF#refs/heads/} = 'master' ] ; then
+          xcrun xccov view --report --json ./output/DerivedData/Logs/Test/*.xcresult > output/coverage/master-coverage.json 2>/dev/null
+          else xcrun xccov view --report --json ./output/DerivedData/Logs/Test/*.xcresult > output/coverage/pullreq-coverage.json 2>/dev/null
+          fi 
+      - name: Upload coverage artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: coverage
+          path: output/coverage
+  
+  coverage:
+    name: Measure Coverage
+    runs-on: macos-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: master
+      - uses: actions/download-artifact@v2
+        with:
+          path: input
+      - name: Code coverage report
+        run: |
+          curl -sL https://github.com/illabo/xcccr/releases/latest/download/xcccr > xcccr
+          chmod +x xcccr
+          if [ ! -f input/coverage/pullreq-coverage.json ] ; then
+          cat input/coverage/master-coverage.json | ./xcccr
+          else 
+          sudo xcode-select --switch /Applications/Xcode_12.app
+          xcodebuild -sdk iphonesimulator -workspace ./FlipperZero/FlipperZero.xcodeproj/project.xcworkspace -scheme 'FlipperZero(iOS)' -derivedDataPath output/DerivedData -enableCodeCoverage YES test -destination platform='iOS Simulator,name=iPhone 11' >/dev/null 2>&1 
```

Comment:
Just wondering is there a way to get the last artifact uploaded for `master` instead of rebuilding `master` for each PR?

Reply:
At first I thought it is possible, but turned out artifacts couldn't be shared between workflows. It is possible to store artifacts in a repository along the code, but it didn't looks as a good practice. Another way around is to store artifacts in some external storage or a separate repository. Or ditch this solution altogether and use, finally, Codecov.io.